ioctl/cmd/ins/insregister.go

```diff
@@ -0,0 +1,352 @@
+          "type": "address"
+        },
+        {
+          "internalType": "bytes[]",
+          "name": "data",
+          "type": "bytes[]"
+        },
+        {
+          "internalType": "bool",
+          "name": "reverseRecord",
+          "type": "bool"
+        },
+        {
+          "internalType": "uint16",
+          "name": "ownerControlledFuses",
+          "type": "uint16"
+        }
+      ],
+      "name": "register",
+      "outputs": [],
+      "stateMutability": "payable",
+      "type": "function"
+    }
+	]`
+)
+
+// _insRegisterCmd represents the action hash command
+var _insRegisterCmd = &cobra.Command{
+	Use:   config.TranslateInLang(_registerUsages, config.UILanguage),
+	Short: config.TranslateInLang(_registerShorts, config.UILanguage),
+	Args:  cobra.MinimumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := register(args)
+		return output.PrintError(err)
+	},
+}
+
+func init() {
+	_insRegisterCmd.Flags().StringVarP(
+		&controller, "controller", "c",
+		"0x8aA6acF9BFeEE0243578305706766065180E68d4",
+		config.TranslateInLang(_registerControllerUsages, config.UILanguage),
+	)
+	_insRegisterCmd.Flags().StringVarP(
+		&resolver, "resolver", "r",
+		"0x41B9132D4661E016A09a61B314a1DFc0038CE3e8",
+		config.TranslateInLang(_registerResolverUsages, config.UILanguage),
+	)
+	_insRegisterCmd.Flags().StringVarP(
+		&owner, "owner", "w",
+		"",
+		config.TranslateInLang(_registerOwnerUsages, config.UILanguage),
+	)
+
+	// init secret
+	sha := sha3.NewLegacyKeccak256()
+	sha.Write([]byte("secret"))
+	hash := sha.Sum(nil)
+	copy(secret[:], hash)
+}
+
+func makeCommitment(
+	name string,
+	owner common.Address,
+	duration *big.Int,
+	secret [32]byte,
+	resolver common.Address,
+	data [][]byte,
+	reverseRecord bool,
+	ownerControlledFuses uint16,
+) ([32]byte, error) {
+	sha := sha3.NewLegacyKeccak256()
+	sha.Write([]byte(name))
+	lable := [32]byte{}
+	copy(lable[:], sha.Sum(nil))
+
+	Bytes32, _ := abi.NewType("bytes32", "", nil)
+	Address, _ := abi.NewType("address", "", nil)
+	Uint256, _ := abi.NewType("uint256", "", nil)
+	BytesArray, _ := abi.NewType("bytes[]", "", nil)
+	Bool, _ := abi.NewType("bool", "", nil)
+	Uint16, _ := abi.NewType("uint16", "", nil)
+	args := abi.Arguments{
+		{Type: Bytes32, Name: "name"},
+		{Type: Address, Name: "owner"},
+		{Type: Uint256, Name: "duration"},
+		{Type: Bytes32, Name: "secret"},
+		{Type: Address, Name: "resolver"},
+		{Type: BytesArray, Name: "data"},
+		{Type: Bool, Name: "reverseRecord"},
+		{Type: Uint16, Name: "ownerControlledFuses"},
+	}
+
+	packed, err := args.Pack(lable, owner, duration, secret, resolver, data, reverseRecord, ownerControlledFuses)
+	if err != nil {
+		return [32]byte{}, err
+	}
+
+	hash := sha3.NewLegacyKeccak256()
+	hash.Write(packed)
+	result := [32]byte{}
+	copy(result[:], hash.Sum(nil))
+
+	return result, nil
+}
+
+func commit(name string, data []byte, ioController address.Address, controllerAbi abi.ABI) error {
```

Comment:
if `ioController` is fixed address, no need to pass in

Reply:
same reason as above